pkg/kubectl/cmd/exec.go

```diff
@@ -195,22 +202,23 @@ func (p *ExecOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, argsIn []
 		}
 	}
 
-	mapper, _ := f.Object()
+	clientMapper := resource.ClientMapperFunc(f.ClientForMapping)
+	mapper, typer := f.Object()
+	decoder := f.Decoder(true)
 
-	gr, name, err := resolveResource(api.Resource(string(api.ResourcePods)), p.PodName, mapper)
+	infos, err := resource.NewBuilder(mapper, typer, clientMapper, decoder).
+		NamespaceParam(p.Namespace).DefaultNamespace().
+		ResourceNames(string(api.ResourcePods), p.PodName).
```

Comment:
not at all, just afraid that it would be out of the patch's scope. It's even stated that such changes should not go with.. but in a separate pr. Will replace and see